src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -569,9 +569,7 @@ private SignatureProvider CreateSignatureProvider(SecurityKey key, string algori
                         signatureProvider = new SymmetricSignatureProvider(key, algorithm, willCreateSignatures);
 
                     if (ShouldCacheSignatureProvider(signatureProvider))
-                        if (CryptoProviderCache.TryAdd(signatureProvider))
-                            Interlocked.Increment(ref signatureProvider._ReferenceCount);
-
+                        CryptoProviderCache.TryAdd(signatureProvider);
```

Comment:
Why do we not need to increment the ref count here?

Reply:
this is the creation case, See Signature Provider ctor, ref count is set to 1 on creation